writer.go

```diff
@@ -132,7 +132,7 @@ func (re *writerResponseEmitter) SetLength(length uint64) {
 		return
 	}
 
-	*re.length = length
+	re.length = &length
```

Comment:
Can you split this in it's own commit in case this SOMEHOW was correct and we want to revert your change ?

Reply:
(It was literally panicking)

Reply:
(But sure)